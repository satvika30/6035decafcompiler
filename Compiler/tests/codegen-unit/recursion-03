class Program {
	int list(int a, int b, int c, int d, int e, int f, int g, int h) {
		callout("printf","list: %d %d %d %d %d %d %d %d\n",a, b, c, d, e, f, g, h);
	
		if(a !=0 ){
			a -= 1;
			return list(a,b,c,d,e,f,g,h);
		}
		
		if(b !=0){
			b -= 1;
			return list(a,b,c,d,e,f,g,h);
		}
		
		if(c != 0){
			c -= 1;
			return list(a,b,c,d,e,f,g,h);
		}
		
		if(d !=0){
			d -= 1;
			return list(a,b,c,d,e,f,g,h);
		}
		
		if(e !=0){
			e -= 1;
			return list(a,b,c,d,e,f,g,h);
		}
		
		if(f !=0){
			f -= 1;
			return list(a,b,c,d,e,f,g,h);
		}
		
		if(g !=0){
			g -= 1;
			return list(a,b,c,d,e,f,g,h);
		}
		
		if(h !=0){
			h -= 1;
			return list(a,b,c,d,e,f,g,h);
		}
		
		return 0;
	}
	
	void main() {
		int k;
		k = list(3,3,3,3,3,3,3,3);
	}
}
