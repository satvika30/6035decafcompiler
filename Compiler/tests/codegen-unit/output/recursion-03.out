list: 3 3 3 3 3 3 3 3
list: 2 3 3 3 3 3 3 3
list: 1 3 3 3 3 3 3 3
list: 0 3 3 3 3 3 3 3
list: 0 2 3 3 3 3 3 3
list: 0 1 3 3 3 3 3 3
list: 0 0 3 3 3 3 3 3
list: 0 0 2 3 3 3 3 3
list: 0 0 1 3 3 3 3 3
list: 0 0 0 3 3 3 3 3
list: 0 0 0 2 3 3 3 3
list: 0 0 0 1 3 3 3 3
list: 0 0 0 0 3 3 3 3
list: 0 0 0 0 2 3 3 3
list: 0 0 0 0 1 3 3 3
list: 0 0 0 0 0 3 3 3
list: 0 0 0 0 0 2 3 3
list: 0 0 0 0 0 1 3 3
list: 0 0 0 0 0 0 3 3
list: 0 0 0 0 0 0 2 3
list: 0 0 0 0 0 0 1 3
list: 0 0 0 0 0 0 0 3
list: 0 0 0 0 0 0 0 2
list: 0 0 0 0 0 0 0 1
list: 0 0 0 0 0 0 0 0
